hummingbot/connector/exchange/coinflex/coinflex_exchange.py

```diff
@@ -0,0 +1,978 @@
+import asyncio
+import logging
+import time
+
+from decimal import Decimal
+from typing import (
+    Any,
+    AsyncIterable,
+    Dict,
+    List,
+    Optional,
+)
+
+from async_timeout import timeout
+
+import hummingbot.connector.exchange.coinflex.coinflex_constants as CONSTANTS
+from hummingbot.connector.client_order_tracker import ClientOrderTracker
+from hummingbot.connector.exchange_base import ExchangeBase
+from hummingbot.connector.exchange.coinflex import coinflex_utils
+from hummingbot.connector.exchange.coinflex.coinflex_api_order_book_data_source import CoinflexAPIOrderBookDataSource
+from hummingbot.connector.exchange.coinflex.coinflex_auth import CoinflexAuth
+from hummingbot.connector.exchange.coinflex.coinflex_http_utils import (
+    api_call_with_retries,
+    build_api_factory,
+    CoinflexAPIError,
+    CoinflexRESTRequest,
+)
+from hummingbot.connector.exchange.coinflex.coinflex_order_book_tracker import CoinflexOrderBookTracker
+from hummingbot.connector.exchange.coinflex.coinflex_user_stream_tracker import CoinflexUserStreamTracker
+from hummingbot.connector.time_synchronizer import TimeSynchronizer
+from hummingbot.connector.trading_rule import TradingRule
+from hummingbot.core.api_throttler.async_throttler import AsyncThrottler
+from hummingbot.core.data_type.cancellation_result import CancellationResult
+from hummingbot.core.data_type.in_flight_order import InFlightOrder, OrderUpdate, OrderState, TradeUpdate
+from hummingbot.core.data_type.limit_order import LimitOrder
+from hummingbot.core.data_type.order_book import OrderBook
+from hummingbot.core.data_type.trade_fee import (
+    DeductedFromReturnsTradeFee,
+    TradeFeeBase,
+)
+from hummingbot.core.event.events import (
+    OrderType,
+    TradeType,
+)
+from hummingbot.core.network_iterator import NetworkStatus
+from hummingbot.core.utils.async_utils import (
+    safe_ensure_future,
+    safe_gather,
+)
+from hummingbot.core.web_assistant.connections.data_types import RESTMethod
+from hummingbot.core.web_assistant.rest_assistant import RESTAssistant
+from hummingbot.logger import HummingbotLogger
+
+s_logger = None
+s_decimal_0 = Decimal(0)
+s_decimal_NaN = Decimal("nan")
+
+
+class CoinflexExchange(ExchangeBase):
+    SHORT_POLL_INTERVAL = 5.0
+    UPDATE_ORDER_STATUS_MIN_INTERVAL = 10.0
+    LONG_POLL_INTERVAL = 120.0
+
+    MAX_ORDER_UPDATE_RETRIEVAL_RETRIES_WITH_FAILURES = 3
+
+    def __init__(self,
+                 coinflex_api_key: str,
+                 coinflex_api_secret: str,
+                 trading_pairs: Optional[List[str]] = None,
+                 trading_required: bool = True,
+                 domain="live"
+                 ):
+        self._domain = domain
+        self._coinflex_time_synchronizer = TimeSynchronizer()
```

Comment:
If a time synchronizer is really required for this exchange there should be code in the update status loop to ensure it regularly receives new server time samples